### PR TITLE
[WL-789] Mobile navigation update

### PR DESCRIPTION
**Pull request by dev-tronghieu**

### CODE REVIEW

File: apps/mobile/src/stacks/Dashboard/TabBar.tsx [hunk 1]
```diff
@@ -59,36 +47,60 @@ export const BottomNavigationTabBar: FC<Props> = ({ tabProps }) => {
...
+					style={{
+						width: iconSize,
+						height: iconSize,
+						borderRadius: iconSize / 2,
+					}}
+					source={{ uri: appState.profile.profileImage }}
+				/>
+			);
+		}
+	}, []);
```
Comment (fluctlight-kayaba) [inline apps/mobile/src/stacks/Dashboard/TabBar.tsx:64]: don't do this, `1.` assign component via hook is bad, `2.` the hook itself run once, which we'll stuck with the initial-rendering even if `appSate.profile` changed.

File: apps/mobile/src/stacks/Dashboard/TabBar.tsx
Comment (fluctlight-kayaba) [inline apps/mobile/src/stacks/Dashboard/TabBar.tsx:1]: please rework this piece

File: apps/mobile/src/stacks/Dashboard/index.tsx [hunk 2]
```diff
@@ -13,8 +17,15 @@ const Tab = createBottomTabNavigator<DashboardParamList>();
...
 
 	const screenOptions: BottomTabNavigationOptions = {
+		header({ route }) {
+			return <Header topInset={insets.top} title={route.name} />;
+		},
+	};
+
+	const noHeader = {
 		headerShown: false,
 	};
```
Comment (fluctlight-kayaba) [inline apps/mobile/src/stacks/Dashboard/index.tsx:30]: put this on separate file, e.g under util

File: apps/mobile/src/stacks/Home/index.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+			return (
+				<Header
+					title={screenName}
+					topInset={insets.top}
+					showIcon={canGoBack}
+					goBack={navigation.goBack}
+				/>
+			);
+		},
+	});
```
Comment (fluctlight-kayaba) [inline apps/mobile/src/stacks/Home/index.tsx:35]: make these kind of logic re-useable

File: apps/mobile/src/stacks/Dashboard/index.tsx (hunk 2 above)
Comment (dev-tronghieu) [reply apps/mobile/src/stacks/Dashboard/index.tsx:30]: please recommend me a name for that file

**Review by fluctlight-kayaba — APPROVED** [review]

**Pull request merged**